routes/index.js

```diff
@@ -0,0 +1,91 @@
+require("dotenv").config();
+const express = require("express");
+const twilio = require("twilio")(process.env.ACCOUNTSID, process.env.AUTHTOKEN);
+const router = express.Router();
+const { signup, factorview, success } = require("../views/");
+const { User, Code } = require("../models");
+
+router.get("/", (_req, res, next) => {
+	try {
+		res.send(signup);
+	} catch (error) {
+		next(error);
+	}
+});
+
+router.post("/", async (req, res, next) => {
+	try {
+		const [user, _created] = await User.findOrCreate({
+			where: {
+				name: req.body.name,
+				email: req.body.email,
+				phone: req.body.phone,
+			},
+		});
+		const validTime = new Date();
+		validTime.setSeconds(validTime.getSeconds() + 30);
+		const code = await Code.create({
+			authCode: Math.floor(1000 + Math.random() * 9000),
+			valid_until: validTime,
+		});
+		await code.setUser(user);
+		const fromNum = process.env.FROMNUM;
+		const toNum = user.phone;
+		const messageBody = "your 2 factor authentication code is " + code.authCode;
+		await twilio.messages.create({
+			body: messageBody,
+			from: fromNum,
+			to: toNum,
+		});
+		await res.redirect(`/auth/${user.id}`);
+	} catch (error) {
+		next(error);
```

Comment:
It'd be nice if there was a descriptive error message.  Or does Express+Sequelize somehow manage one?  What happens if you try to add a user twice?

Reply:
the default error message doesn't read well, you can customize the error message in the unique field in the model pretty easily i believe